tests/integration.rs

```diff
@@ -5712,6 +5720,99 @@ mod tests {
 
             handle_child_output(r, &output);
         }
+
+        #[test]
+        fn test_windows_guest_cpu_hotplug() {
+            let resize_to_num = 6;
+
+            let mut windows = WindowsDiskConfig::new(WINDOWS_IMAGE_NAME.to_string());
+            let guest = Guest::new(&mut windows);
+
+            let tmp_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut ovmf_path = workload_path.clone();
+            ovmf_path.push(OVMF_NAME);
+
+            let mut osdisk_path = workload_path;
+            osdisk_path.push(WINDOWS_IMAGE_NAME.to_string());
+
+            let api_socket = temp_api_path(&tmp_dir);
+
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=2,max=8,kvm_hyperv=on"])
+                .args(&["--memory", "size=4G"])
+                .args(&["--kernel", ovmf_path.to_str().unwrap()])
+                .default_disks()
+                .args(&["--serial", "tty"])
+                .args(&["--console", "off"])
+                .args(&["--net", "tap="])
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            // Wait to make sure Windows boots up
+            thread::sleep(std::time::Duration::new(60, 0));
+
+            let r = std::panic::catch_unwind(|| {
+                let auth = windows_auth();
+
+                // Issue hotadd CPUs
+                assert!(remote_command2(
```

Comment:
I would suggest the whole sequence to be:
- Check the number of vcpus is 2
- Resize to 6
- Check the number of vcpus is 6
- Resize to 4
- Check the number of vcpus is 4

This would validate:
- the VM is started with the right amount of vCPUs
- hotplugging vcpus work
- unplugging vcpus work as well

Reply:
Yeah, that's a good plan. I'll rework with the suggested sequence.

Thanks

Reply:
Cool! Thanks @weltling

Reply:
I refactored the test based on the suggestions. Thanks to this, an issue with hot-remove is revealed. Checking on what was there in #1799 originally, seems remove was not really the topic. I guess this will need yet another investigation round :(, the hot-remove part is commented out in the test right now.

Thanks